src/components/WorkspaceDashboardTaskRunsCard.vue

```diff
@@ -0,0 +1,199 @@
+    if (isDefined(completed.value) && isDefined(total.value)) {
+      return toPercent(completed.value, total.value)
+    }
+
+    return undefined
+  })
+
+  const failedTasksFilter = computed<TaskRunsFilter>(() => ({
+    taskRuns: {
+      ...tasksFilter.value.taskRuns,
+      state: {
+        type: ['FAILED', 'CRASHED'],
+      },
+    },
+  }))
+  const failedTasksSubscription = useSubscription(api.taskRuns.getTaskRunsCount, [failedTasksFilter])
+  const failed = computed(() => failedTasksSubscription.response)
+  const failedPercentage = computed(() => {
+    if (isDefined(failed.value) && isDefined(total.value)) {
+      return toPercent(failed.value, total.value)
+    }
+
+    return undefined
+  })
+
+  const historyFilter = computed(() => mapper.map('WorkspaceDashboardFilter', props.filter, 'TaskRunsHistoryFilter'))
+  const historySubscription = useSubscription(api.taskRuns.getTaskRunsHistory, [historyFilter])
+  const history = computed(() => historySubscription.response ?? [])
+
+  const taskRunsChartData = computed(() => {
+    const completed: LineChartData = []
+    const failed: LineChartData = []
+
+    history.value.forEach(item => {
+      let completedCount = 0
+      let failedCount = 0
+
+      item.states.forEach(state => {
+        if (state.stateType === 'COMPLETED') {
+          completedCount += state.countRuns
+        } else if (['FAILED', 'CRASHED'].includes(state.stateType)) {
+          failedCount += state.countRuns
+        }
+      })
+
+      const [, completedBase = 0] = completed.at(-1) ?? []
+      const [, failedBase = 0] = failed.at(-1) ?? []
+
+      completed.push([item.intervalStart, completedBase + completedCount])
+      failed.push([item.intervalStart, failedBase + failedCount])
+    })
+
+    return {
+      completed,
+      failed,
+    }
+  })
+
+  const maxFailedValue = computed(() => {
+    const values = taskRunsChartData.value.failed.map(([, y]) => y)
+    const max = Math.max(...values)
+
+    if (max === 0) {
+      return undefined
+    }
+
+    return max * 4
+  })
+
+  function toPercent(x: number, y: number): string | undefined {
+    const decimal = x / y
+    const percent = Math.round((decimal + Number.EPSILON) * 10000) / 100
+
+    if (isNaN(percent)) {
+      return undefined
+    }
+
+    return `${percent}%`
+  }
+</script>
+
+<style>
+.workspace-dashboard-task-runs-card { @apply
+  relative
+  grid
+  gap-4
+  overflow-hidden
+}
+
+.workspace-dashboard-task-runs-card__summary { @apply
+  grid
+  gap-1
+}
+
+.workspace-dashboard-task-runs-card__statistic--completed .dashboard-statistic__value { @apply
+  text-state-completed-600
+}
+
+.workspace-dashboard-task-runs-card__statistic--failed .dashboard-statistic__value { @apply
+  text-state-failed-700
+}
+
+.workspace-dashboard-task-runs-card__chart-container { @apply
+  relative
+  min-h-0
+  h-16
+}
+
+.workspace-dashboard-task-runs-card__chart { @apply
+  absolute
+  min-h-0
+  -left-8
+  top-0
+  -right-8
+  bottom-0
```

Comment:
I think we'll end up wanting some `bottom` amount, but we can do that when we're getting the actual layout together and see it next to the events card, etc.

Reply:
This bottom is relative to the container div which does respect the cards padding. So this is full width but still 16px from the bottom of the card.

Reply:
Ah I see in the screenshot, ty